rust/lance-encoding/src/encoder.rs

```diff
@@ -408,7 +416,10 @@ impl CoreBufferEncodingStrategy {
             return None;
         }
 
-        Some(BitpackingBufferEncoder::new(num_bits))
+        Some(BitpackingBufferEncoder::new(
+            num_bits,
+            !data_type.is_unsigned_integer(),
```

Comment:
Not a huge deal but we could probably get away with only adding the sign bit if there are actually negative values in the page.  For example, a user might use i32 because they think they might need support for negatives someday but then never actually store any negative values.

Reply:
Good call. I can add this in a follow-up